Sources/SwiftDiscord/Gateway/DiscordSharding.swift

```diff
@@ -183,19 +183,20 @@ open class DiscordShardManager : DiscordShardDelegate, Lockable {
         **Note** This method is an async method.
     */
     open func connect() {
-        func _connect() {
-            let shards = get(self.shards)
-
-            protected { closed = false }
-
-            for shard in shards where get(!closed) {
-                shard.connect()
-
-                Thread.sleep(forTimeInterval: 5.0)
+        func _connect(iterator: IndexingIterator<[DiscordShard]>) {
+            var iterator = iterator
+
+            guard get(!closed) else { return }
+            guard let shard = iterator.next() else { return }
+            shard.connect()
+            
+            DispatchQueue.global().asyncAfter(deadline: DispatchTime(secondsFromNow: 5)) {
```

Comment:
Looking at this now, there's really nothing stopping this from just firing _N_ `.asyncAfter` calls. That way it could avoid the `var iterator = iterator` anti-pattern.